File: apps/android/core/network/src/main/java/com/scribblefit/core/network/model/AuthModels.kt
package com.scribblefit.core.network.model

import kotlinx.serialization.Serializable

@Serializable
data class AuthRequest(
    val deviceId: String,
    val subscriptionId: String? = null
)

@Serializable
data class AuthResponse(
    val token: String,
    val expiresAt: Long
)

File: apps/ios/ScribbleFit/Sources/Core/Network/Models/AuthModels.swift
import Foundation

public struct AuthRequest: Codable, Sendable {
    public let deviceId: String
    public let subscriptionId: String?
    
    public init(deviceId: String, subscriptionId: String? = nil) {
        self.deviceId = deviceId
        self.subscriptionId = subscriptionId
    }
}

public struct AuthResponse: Codable, Sendable {
    public let token: String
    public let expiresAt: Int64
    
    public init(token: String, expiresAt: Int64) {
        self.token = token
        self.expiresAt = expiresAt
    }
}
